### PR TITLE
Tying backend API and frontend for COR and D6T submissions

### LINKED ISSUE #228: Backend - Bulk submission - POST for D6Ts
**Is your feature request related to a problem? Please describe.**
As a user, I want to press a button and have a single D6T submit

**Describe the solution you'd like**
Bulk submission on the D6T page and have the backend queue up each status creation and post them one at a time.

**Describe alternatives you've considered**
Single submission vice bulk submission

**Additional context**
NA

### LINKED ISSUE #229: Backend - Bulk submission - POST for COR
**Is your feature request related to a problem? Please describe.**
As a user, I want to press a button and have bulk COR submission

**Describe the solution you'd like**
Bulk submission on the COR page and have the backend queue up each status creation and post them one at a time.

**Describe alternatives you've considered**
Single submission vice bulk submission

**Additional context**
NA

### PR DESCRIPTION
**Pull request by dds-hansolo**
This issue spans multiple issues have been addressed in displaying and commit for bulk d6t and cor submissions.

Thanks for submitting a pull request! Below are a few things you can do to help us more quickly review your changes.
closes #228 
closes #229

## Checklist
I have…
- [X] summarized below my changes and noted which issues (if any) this pull request fixes or addresses.
- [X] thoroughly outlined below the steps necessary to test my changes.
- [X] attached screenshots illustrating relevant behavior before and after my changes.
- ~~[ ] review and update SSAG documentation if needed.~~

## Summary of Changes
This pull request…
- Pagination of api requests.
- Bulk API endpoints for D6T/COR submissions

## Testing
To verify the changes proposed in this pull request…
1. Generate COR/D6T POST requests. Also see the unit tests.

## Screenshots
N/A.

### CODE REVIEW

**Comment by lit-dds** [comment]
Missing frontend files to finish the connection.

File: backend/solo_rog_api/urls.py [hunk 1]
```diff
@@ -7,4 +7,7 @@
     path("login/refresh/", TokenRefreshView.as_view(), name="login_refresh"),
     path("workerlog/", views.CeleryDebugTaskView.as_view(), name="workerlog"),
     path("document/", views.DocumentList.as_view(), name="document_list"),
+    path("document/d6t/", views.D6TSubmissionView.as_view(), name="bulk_d6t"),
+    path("document/cor/", views.CORSubmissionView.as_view(), name="bulk_cor"),
+    path("document/cor/bulk/", views.CORSubmissionView.as_view(), name="bulk_cor"),
```
Comment (jhackshaw-dds) [inline backend/solo_rog_api/urls.py]: can we just get rid of the /cor/bulk/ endpoint since it's the same?

File: backend/solo_rog_api/views.py [hunk 2]
```diff
@@ -46,3 +53,35 @@ class DocumentList(generics.ListAPIView):
...
+        self, *args: Any, **kwargs: Any
+    ) -> Union[BaseSerializer, BaseSerializer]:
+        return super().get_serializer(many=True, allow_empty=False, **kwargs)
+
+    def get_serializer_context(self) -> Dict[str, Any]:
+        ctx = super().get_serializer_context()
+        ctx["new_status"] = "D6T"
+        ctx["document_excludes"] = {"statuses__dic__code": "D6T"}
+        ctx["document_filters"] = {"statuses__dic__code": "AS2"}
+        return ctx
```
Comment (jhackshaw-dds) [inline backend/solo_rog_api/views.py]: This shouldn't be necessary now that the COR/D6T submit serializers are separate

File: backend/solo_rog_api/views.py [hunk 3]
```diff
@@ -46,3 +53,35 @@ class DocumentList(generics.ListAPIView):
...
+        self, *args: Any, **kwargs: Any
+    ) -> Union[BaseSerializer, BaseSerializer]:
+        return super().get_serializer(many=True, allow_empty=False, **kwargs)
+
+    def get_serializer_context(self) -> Dict[str, Any]:
+        ctx = super().get_serializer_context()
+        ctx["new_status"] = "COR"
+        ctx["document_excludes"] = {"statuses__dic__code": "COR"}
+        ctx["document_filters"] = {"statuses__dic__code": "D6T"}
+        return ctx
```
Comment (jhackshaw-dds) [inline backend/solo_rog_api/views.py]: This shouldn't be necessary now that the COR/D6T submit serializers are separate

File: frontend/src/pages/ConfirmationOfReceiptPage/useCORDocuments.ts [hunk 4]
```diff
@@ -45,11 +45,11 @@ const useCORDocuments = () => {
...
         // submit single document and remove from current list
-        await apiCall("/document/cor", {
+        await apiCall("/document/cor/", {
           method: "POST",
           body: JSON.stringify({
             sdn,
-            receivedBy,
+            received_by,
             status: "COR"
           })
```
Comment (jhackshaw-dds) [inline frontend/src/pages/ConfirmationOfReceiptPage/useCORDocuments.ts]: I think this needs to send an array, something like
```
apiCall('endpoint', {
    method: 'POST',
    body: JSON.stringify([{
        sdn,
        received_by,
    }])
)
```

File: frontend/src/pages/ConfirmationOfReceiptPage/__tests__/submitCOR.spec.tsx [hunk 5]
```diff
@@ -77,12 +77,12 @@ describe("ConfirmationOfReceiptPage component submit COR functionality", () => {
...
-      expect(fetchMock.mock.calls[1][0]).toEqual("/document/cor");
+      expect(fetchMock.mock.calls[1][0]).toEqual("/document/cor/");
       expect(fetchMock.mock.calls[1][1]).toMatchObject({
         method: "POST",
         body: JSON.stringify({
           sdn: defaultDoc.sdn,
-          receivedBy: "Private Casarotto, Stuart",
+          received_by: "Private Casarotto, Stuart",
           status: "COR"
         })
```
Comment (jhackshaw-dds) [inline frontend/src/pages/ConfirmationOfReceiptPage/__tests__/submitCOR.spec.tsx]: this test would need to be updated if we're always submitting an array

**Review by jhackshaw-dds — CHANGES_REQUESTED** [review]

**Review by jhackshaw-dds — APPROVED** [review]

**Pull request merged**